### DISCUSSION TITLE
Aggregating Streams after StreamProvider.stream deprecation

### DISCUSSION THREAD

**Discussion by timcreatedit** [rrousselGit/riverpod#2603] [Q&A]
I'm wondering how to implement something like the chat example in the docs (https://riverpod.dev/docs/providers/stream_provider) but in my case, I'm not listening to a socket but another `StreamProvider`. Since `.stream` is deprecated now, the idea I came up with is this:

```dart
final messageProvider = StreamProvider<String>(...);

@riverpod
class Chat extends _$Chat {
  @override
  List<String> build() {
    ref.listen(messageProvider, (_, next) {
      if (next case AsyncData(value: final message?)) {
        state = [...state, message];
      }
    });
    return [];
  }
}
```

This solution seems quite cumbersome compared to the one using `.stream`, so I was wondering if there's something I'm missing. So far, I'm struggling a little bit to see the reason for the `.stream` deprecation, since I keep running into things like this, but maybe my mental model isn't there yet.
Thanks for any help!

**Comment by rskvazh** [comment]
Same question. I have a REST API with last messages, and need to update list of messages from websocket connection.

**Comment by rrousselGit** [comment]
Using ref.listen as you did is fine.

Although note that you may want to implement a custom Stream extension instead. Converting a Stream<T> into a Stream<List<T>>  could be done with a custom `stream.combined` or something
